soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,174 @@
+                        0.5, 0.9, 100, 1, 0.1)#AdaptiveFormation.NELDER_MEAD_ARGS
+
+    class State(enum.Enum):
+        ## getting ready to recieve a pass from another robot
+        passSet = 1
+
+        hold = 2
+        
+
+    def __init__(self):
+        super().__init__(continuous=True)
+
+        #Sends one robot to best reciever point, and other to the best receiving point
+        # of the first midfielder
+        self.add_state(AdvanceZoneMidfielder.State.passSet,
+                       behavior.Behavior.State.running)
+
+        self.add_state(AdvanceZoneMidfielder.State.hold, 
+                        behavior.Behavior.State.running)
+
+        self.moves = [None, None]
+
+
+        self.priorities = [1, 2]
+
+        self.passing_point = main.ball().pos
+
+        y_temp_hold = 0.8 * self.passing_point.y
+
+        x_temp_hold = constants.Field.Width / 3
+
+        self.hold_point = [robocup.Point(x_temp_hold,y_temp_hold), robocup.Point(x_temp_hold,y_temp_hold)]
+
+        self.dribble_state = True
+
+        self.kick = False
+
+        for s in AdvanceZoneMidfielder.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        self.names = ['left','right']
+        #intially should be passSet method
+        self.add_transition(behavior.Behavior.State.start,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "Immediately")
+        self.add_transition(AdvanceZoneMidfielder.State.passSet,
+                            AdvanceZoneMidfielder.State.hold, lambda: False,
+                            "When a kick play begins")
+        self.add_transition(AdvanceZoneMidfielder.State.hold,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "When not in a kick play")
+
+    def on_enter_passSet(self):
+        print("Entering PassSet");
+
+    def execute_passSet(self):
+        #gets the best position to travel to for ball reception
+        best_Pos_From_Ball = self.passing_point
+        """if (self.dribble_state):
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        else :
+            points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+            self.optimal = 1
+            # sets either the left or the right side of the ball to the best ball position
+            # and then notes which one is the other
+            # left side of the ball
+            points[self.optimal] = best_Pos_From_Ball
+            self.other = 0
+        """
+        points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+        points[0] = best_Pos_From_Ball;
```

Comment:
You should be able to just initialize points with `self.passing_point` already in the first position. This will remove the need for `best_Pos_From_Ball`